packages/datatrak-web/src/features/Survey/SurveyQuestion.tsx

```diff
@@ -24,51 +35,44 @@ const Placeholder = ({ name, type, id }) => {
   );
 };
 
-const InstructionQuestion = ({ text }) => {
-  return <QuestionPlaceholder>{text}</QuestionPlaceholder>;
-};
-
 export enum QUESTION_TYPES {
-  Binary = Placeholder,
-  Checkbox = Placeholder,
-  Date = Placeholder,
-  DateTime = Placeholder,
-  FreeText = TextField,
+  Binary = BinaryQuestion,
+  Checkbox = CheckboxQuestion,
+  Date = DateQuestion,
+  DateTime = DateTimeQuestion,
+  FreeText = TextQuestion,
   Geolocate = Placeholder,
   Autocomplete = Placeholder,
   Instruction = InstructionQuestion,
-  Number = TextField,
+  Number = TextQuestion,
   Photo = Placeholder,
-  Radio = Placeholder,
-  DaysSince = Placeholder,
-  MonthsSince = Placeholder,
-  YearsSince = Placeholder,
-  SubmissionDate = Placeholder,
-  DateOfData = Placeholder,
+  Radio = RadioQuestion,
+  SubmissionDate = DateQuestion,
+  DateOfData = DateQuestion,
   Entity = Placeholder,
   PrimaryEntity = Placeholder,
   CodeGenerator = Placeholder,
   Arithmetic = Placeholder,
   Condition = Placeholder,
 }
 
-interface SurveyQuestionProps {
-  type: keyof typeof QUESTION_TYPES;
-  name: string;
-  id: string;
-  register?: any;
-  label?: string;
-  code?: string;
-  text?: string;
-  options?: any;
-  config?: any;
-}
-export const SurveyQuestion = (props: SurveyQuestionProps) => {
-  const FieldComponent = QUESTION_TYPES[props.type];
+export const SurveyQuestion = ({ type, name, ...props }: SurveyQuestionFieldProps) => {
+  const { register, control } = useFormContext();
+  const FieldComponent = QUESTION_TYPES[type];
 
   if (!FieldComponent) {
-    return <QuestionPlaceholder>{props.name}</QuestionPlaceholder>;
+    return <QuestionPlaceholder>{name}</QuestionPlaceholder>;
   }
 
-  return <FieldComponent inputRef={props.register()} {...props} />;
+  // Use a Controller so that the fields that require change handlers, values, etc work with react-hook-form, which is uncontrolled by default
+  return (
+    <Controller
+      name={name!}
+      control={control}
+      render={renderProps => (
+        <FieldComponent {...props} controllerProps={renderProps} name={name} type={type} />
+      )}
```

Comment:
```suggestion
      render={renderProps => {
       const onChange = e => {
          setFormData({ ...formData, [name]: e.target.value });
          renderProps.onChange(e);
        };
        return <FieldComponent {...props} controllerProps={{ ...renderProps, onChange }} name={name} type={type} />
      }}
```
Since we will need to support navigating between questions, we will also have to hook up the on change handler to keep the form state up to date - something like this. We could do it later but might as well include it now.

Reply:
I'm pretty sure we don't need to add an onChange handler to set the form data, since they're now controlled inputs the value gets passed through with the controller and persists between screens